packages/lesswrong/lib/SwitchingCollection.ts

```diff
@@ -0,0 +1,259 @@
+import { MongoCollection } from "./mongoCollection";
+import PgCollection from "./sql/PgCollection";
+import { getCollectionLockType, setCollectionLockType } from "./mongo2PgLock";
+
+export type ReadTarget = "mongo" | "pg";
+export type WriteTarget = ReadTarget | "both";
+
+const POLL_RATE_SECONDS = 5;
```

Comment:
Relevant to my comments below about the possibility of data not being copied over, we may as well make this shorter if there's really very little overhead. We only have about 15 req/s so making it 1 or 2 seconds gets the potential for getting into a weird state pretty low

Reply:
Fair point - I'll set it to 1 second for now which should be fine. Even 0.5 seconds might be ok given that the db and the server are in the same data centre 🤔.